Files/Filesystem/StorageEnumerators/UniversalStorageEnumerator.cs

```diff
@@ -186,71 +188,80 @@ CancellationToken cancellationToken
+                            itemThumbnailImgVis = false;
+                        }
                     }
-                    else
+                    catch
                     {
                         itemEmptyImgVis = true;
                         itemThumbnailImgVis = false;
+                        // Catch here to avoid crash
                     }
                 }
-                catch
-                {
-                    itemEmptyImgVis = true;
-                    itemThumbnailImgVis = false;
-                    // Catch here to avoid crash
-                }
-            }
-            else
-            {
-                try
+                else
                 {
-                    using var itemThumbnailImg = suppressThumbnailLoading ? null :
-                        await file.GetThumbnailAsync(ThumbnailMode.ListView, 80, ThumbnailOptions.UseCurrentScale);
-                    if (itemThumbnailImg != null)
+                    try
                     {
-                        itemEmptyImgVis = false;
-                        itemThumbnailImgVis = true;
-                        icon.DecodePixelWidth = 80;
-                        icon.DecodePixelHeight = 80;
-                        await icon.SetSourceAsync(itemThumbnailImg);
-                        iconData = await itemThumbnailImg.ToByteArrayAsync();
+                        using var itemThumbnailImg = suppressThumbnailLoading ? null :
+                            await file.GetThumbnailAsync(ThumbnailMode.ListView, 80, ThumbnailOptions.UseCurrentScale);
+                        if (itemThumbnailImg != null)
+                        {
+                            itemEmptyImgVis = false;
+                            itemThumbnailImgVis = true;
+                            icon.DecodePixelWidth = 80;
+                            icon.DecodePixelHeight = 80;
+                            await icon.SetSourceAsync(itemThumbnailImg);
+                            iconData = await itemThumbnailImg.ToByteArrayAsync();
+                        }
+                        else
+                        {
+                            itemEmptyImgVis = true;
+                            itemThumbnailImgVis = false;
+                        }
                     }
-                    else
+                    catch
                     {
                         itemEmptyImgVis = true;
                         itemThumbnailImgVis = false;
                     }
                 }
-                catch
-                {
-                    itemEmptyImgVis = true;
-                    itemThumbnailImgVis = false;
-                }
+
+                return (icon, iconData, itemThumbnailImgVis, itemEmptyImgVis);
             }
+
             if (cancellationToken.IsCancellationRequested)
             {
                 return null;
             }
 
+            await CoreApplication.MainView.Dispatcher.YieldAsync(Windows.UI.Core.CoreDispatcherPriority.Normal);
```

Comment:
Could you explain me this one? I understand that now ListEntries is executed on a background thread, what's the point of yielding to the UI thread?

Reply:
@hez2010 I've noticed that the code to load thumbail inside `AddFileAsync` was never called (the function is always ever called with `suppressThumbnailLoading=true`) and removed it. Please let me know if you agree with the changes!

Reply:
BitmapImage need to be constructed in main thread, so I yield first to switch to main thread.

Reply:
LGTM